src/main/kotlin/org/rust/lang/core/macros/MacroExpansionTask.kt

```diff
@@ -101,6 +103,22 @@ class MacroExpansionTask(
         val rangeMap: RangeMap,
     )
 
+    private data class PreparedFileDeletionAndCreation1(
+        val creation: FileCreation,
+        val oldPath: String,
+        val newParentPath: String,
+        val newName: String,
+    )
+
+    private data class PreparedFileDeletionAndCreation2(
+        val creation: FileCreation,
+        val newName: String,
+        val oldFile: VirtualFile,
+        val oldPsiFile: PsiFile,
+        val nearestNewParentFile: VirtualFile,
+        val segmentsToCreate: List<String>,
+    )
```

Comment:
Maybe move these classes nearer to the `applyChangesFastPath` function

Reply:
I think I should move all theses classes (FileCreation, FileDeletion, FileAttributes, PreparedFileDeletionAndCreation1, PreparedFileDeletionAndCreation2) to the end of the class MacroExpansionTask. I guess I just forgot to move them previously (in #8977)